### PR TITLE
Users able to accept the group invitation on clicking group title. (without clicking on Accept button)

### LINKED ISSUE #1024: Users able to accept the group invitation on clicking group title. (without clicking on Accept button)
**Describe the bug**
If any user has been invited to private group, then it shows two button ( Accept and Reject ) after clicking on notification. But user is able to accept the group invitation on clicking group title. (without clicking on Accept button) 

**To Reproduce**
1. Go to 'Group single page and send invitation to a user'
2. Then login as invitee user and click on invite notification
3. It will go to a page where there is a group name/link & accept/reject button.
4. If user click on group link, he becomes a member of that group, no need to click on accept button

**Expected behavior**
User should not become a member after click on group link, must have to press accept button

**Screenshots**
In this video, i showed how default wp/bp works and how buddyboss-platform/buddyboss-theme works - https://www.loom.com/share/d05141197761435980500abd17d2afce

**Support ticket links**
n/a

### PR DESCRIPTION
**Pull request by KartikSuthar**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1024 .

### How to test the changes in this Pull Request:

1. Go to 'Group single page and send an invitation to a user'
2. Then log-in as invitee user and click on invite notification
3. It will go to a page where there is a group name/link & accept/reject button.
4. If user click on group link, he becomes a member of that group, no need to click on the accept button

### Proof Screenshots or Video
http://somup.com/cY1D6FQr8Q

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/buddypress/groups/single/request-membership.php [hunk 1]
```diff
@@ -6,9 +6,80 @@
...
+					<?php endif; ?>
+
+					<div class="item">
+
+						<div class="item-block">
+
+							<h2 class="list-title groups-title"><?php bp_group_link(); ?></h2>
+
+							<p class="item-meta group-details">
+								<?php $inviter = bp_groups_get_invited_by( bp_loggedin_user_id(), $bp->groups->current_group->id ); ?>
```
Comment (mehulkaklotar) [inline src/bp-templates/bp-nouveau/buddypress/groups/single/request-membership.php:36]: @KartikSuthar can we use `bp_get_current_group_id()` instead?

Comment (KartikSuthar) [reply src/bp-templates/bp-nouveau/buddypress/groups/single/request-membership.php]: @mehulkaklotar used function instead of a global variable.

**Comment by mohit-buddyboss** [comment]
@KartikSuthar  

**Private Group Invitation :** When user clicks on Group title then user gets navigates to group page where user gets Request Access Tab which displays Accept and Reject Button . User also have Accept Invitation button on group page. 

**Issue 1 :** When user clicks on "Accept Invitation" button then nothing is happening. 
Expected : On click of "Accept Invitation" button user should become member of Group and gets navigate to groups members listing tab.

**Issue 2 :** When user clicks on "Accept' Button  exist in Request tab of Group page then user gets navigate to 404 page instead of this user must get navigate to Group members listing page. 

For better understanding see this Video : https://www.loom.com/share/f5064a80122840198c6450cea2636eda  

cc @chetansatasiya @ankitguptaindia @eisenwasser

**Comment by KartikSuthar** [comment]
@mohit-buddyboss As you have seen the multiple accept invitations buttons on the same page. 
We have thought that to place a simple notice there for the user. 
Where you can see the information about the invitation and link to the Group Invites page.
http://prntscr.com/t2o7s8

**Comment by mohit-buddyboss** [comment]
> @mohit-buddyboss As you have seen the multiple accept invitations buttons on the same page.
> We have thought that to place a simple notice there for the user.
> Where you can see the information about the invitation and link to the Group Invites page.
> http://prntscr.com/t2o7s8

@KartikSuthar Yes, Notice is good. But can we show 'Reject invitation' button also here with notice so that in case if user wants to Reject invitation then he/she doesn't can do that from the same page .

**Comment by mehulkaklotar** [comment]
> > @mohit-buddyboss As you have seen the multiple accept invitations buttons on the same page.
> > We have thought that to place a simple notice there for the user.
> > Where you can see the information about the invitation and link to the Group Invites page.
> > http://prntscr.com/t2o7s8
> 
> @KartikSuthar Yes, Notice is good. But can we show 'Reject invitation' button also here with notice so that in case if user wants to Reject invitation then he/she can do that from the same page. cc @chetansatasiya @ankitguptaindia @eisenwasser

@mohit-buddyboss Request Tab for group is visible for the user to see or check if user has Pending invite or not so we don't need to show any buttons to accept or reject invite in the Request tab. That doesn't make sense. There is a View Invitation link which can lead user to proper invitation where user can accept or reject. And plus we are showing Accept Invitation button on the group header as well. So it makes sense to only show notice in the Request tab to user that user already has an invite pending for that group.

According to me this is done and should be merged immediately. We don't want to delay this more.

**Comment by mohit-buddyboss** [comment]
@mehulkaklotar Can you please reopen this issue ? I have verified it and I'm getting Parse Error also Private Group Label is appearing two times and Accept Invitation button is also not working. 
See this Video : https://www.loom.com/share/ca4f4471c3654597be369214a0c82512
Screenshot : https://prnt.sc/t478b7

cc @KartikSuthar

**Comment by mehulkaklotar** [comment]
@mohit-buddyboss I don't get it at all. Please see this video. https://www.loom.com/share/2ecc4febc03946738c30fdb09d4cff4a

**Comment by mohit-buddyboss** [comment]
@mehulkaklotar As per discussion after latest fix this issue is resolve now. For Proof Kindly see this video : https://www.loom.com/share/1afbd75bdaeb4c0ebe5acfea7582cc2d

Thanks
cc @KartikSuthar

**Pull request merged**